doc/source/whatsnew/v2.2.0.rst

```diff
@@ -97,6 +97,17 @@ This future dtype inference logic can be enabled with:
 Enhancements
 ~~~~~~~~~~~~
 
+.. _whatsnew_220.enhancements.strftime:
+
+Strftime performance improvements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+- New :func:`convert_strftime_format` to convert a strftime formatting template into a python string formatting template.
+- New :meth:`Period.fast_strftime` and :meth:`Timestamp.fast_strftime` leveraging templates created with :func:`convert_strftime_format`
```

Comment:
@MarcoGorelli this relates to your comment. As you can see even if the `fast_strftime` argument is now hidden and automatically done whenever possible, the `convert_strftime_format` tool is still public (for maintainance/debugging/development purposes) and therefore `Period.fast_strftime` and :meth:`Timestamp.fast_strftime` can stay public for advanced users. This does not change the fact that these are used in a fully automated way whenever possible.

Reply:
thanks - OK should have capacity to get to this next week, let's see if we can get this in

Reply:
That would be awesome. For the instance-related `Period.fast_strftime` and `Timestamp.fast_strftime` we have three options:

 1. current version "as is"
 2. adding an underscore prefix as mentioned above
 3. dropping these methods and inlining them instead. Not sure that it would bring something, but it would certainly imply to modify some of my tests.